zowe/zowe-cli-sample-plugin
# Add lifecycle example to sample plugin

**What It Does**



A new class implements the PluginLifeCycle class, with documentation and example techniques.

The old pluginHealthCheck is removed.

**How to Test**



- Mostly review the Readme and the comments in LifeCycleForSample.ts.

- Install the sample plugin and see the new postInstall message that it displays.
    ```
    > zowe plugins install C:\ourstuff\repos\plugins\zowe-cli-sample-plugin
    Plug-ins within the Imperative CLI Framework can legitimately gain
    control of the zowe CLI application during the execution of every command.
    Install 3rd party plug-ins at your own risk.

    Registry = https://registry.npmjs.org/

    _______________________________________________________________
    [2023/05/08 13:05:33.662] [WARN] [LifeCycleForSample.js:63] Something weird happened in the sample plugin's postInstall function. However, things should still work ok.
    Installed plugin name = '@zowe/sample-plugin-for-zowe-cli'

    _____ Validation results for plugin '@zowe/sample-plugin-for-zowe-cli' _____
    This plugin was successfully validated. Enjoy the plugin.
    ```

- Uninstall the plugin and see the new preUninstall message that it displays.
    ```
    > zowe plugins uninstall @zowe/sample-plugin-for-zowe-cli
    [2023/05/08 13:07:32.802] [WARN] [LifeCycleForSample.js:113] Something weird happened in the sample plugin's preUninstall function that we thought you should know about.
    Removal of the npm package(s) was successful.
    ```

**Review Checklist**

I certify that I have:





**Additional Comments**



Now that the functionality is complete, I will attempt to update eslint to a newer version which does not get a warning that it is unsupported with the version of typescript being used. Since that change is beyond the scope of this story, I will abandon the effort if it proves too time-consuming.
